src/main/java/org/example/bookstore/security/CustomUserDetailsService.java

```diff
@@ -0,0 +1,20 @@
+package org.example.bookstore.security;
+
+import lombok.RequiredArgsConstructor;
+import org.example.bookstore.repository.UserRepository;
+import org.springframework.security.core.userdetails.UserDetails;
+import org.springframework.security.core.userdetails.UserDetailsService;
+import org.springframework.security.core.userdetails.UsernameNotFoundException;
+import org.springframework.stereotype.Service;
+
+@Service
+@RequiredArgsConstructor
+public class CustomUserDetailsService implements UserDetailsService {
+    private final UserRepository userRepository;
+
+    @Override
+    public UserDetails loadUserByUsername(String email) throws UsernameNotFoundException {
+        return userRepository.findByEmail(email).orElseThrow(() -> new UsernameNotFoundException(
+                "Can`t" + " find user by email"));
```

Comment:
Maybe you should add <"by email: " + email>

Reply:
yes, thanks, actually it looks a bit strange, there is `+` but no email)